### ISSUE TITLE
recurring: Unexpected Data Format during capturing Instagram Data

### ISSUE THREAD

**Issue by bodilvreugdenhil** [digitalmethodsinitiative/zeeschuimer#29]
Hi!

It seems as though this exact problem has come up again:

https://github.com/digitalmethodsinitiative/zeeschuimer/issues/27

I experienced the same error as this user when trying to capture data from Instagram.
I did make sure that 4CAT and Zeeschuimer were completely up to date. Perhaps Instagram changed their data structure again recently.

Thanks in advance!

**Comment by dale-wahl** [comment]
Could you check the dataset log in 4CAT. Is the error the same as in this part: `Unable to map item: KeyError-'full_name'`? Or are you receiving the unexpected data format warning for a different key name?

**Comment by bodilvreugdenhil** [comment]
Yes, it is exactly that error. Unable to map item: KeyError-'full_name'

**Comment by dale-wahl** [comment]
Ah, the fix did not make the 1.40 release, but is available in the master version of 4CAT (https://github.com/digitalmethodsinitiative/4cat/commit/fa3be93bafef17e95881207604efa1212d562d9e).

You can use the “latest” Docker tag in the .env file to use the most current version of 4cat which I checked and includes the above fix. After updating 4cat, the status message of the Dataset will not update (unless you re-import the data), but all processors will use the `author_fullname` field appropriately.

**Comment by namansnghl** [comment]
Hi, was this resolved? I see this problem came back and even the latest version does not solve it. Am I missing something?

`"MapItemException (item 1): Unable to map item: KeyError-'like_count'"`

**Comment by dale-wahl** [comment]
@namansnghl the original error with `full_name` was resolved and released. 

Your error appears to be a different data issue which means Instagram reformatted their data again and someone will have to take a look.

**Comment by namansnghl** [comment]
@dale-wahl I will raise an issue on 4cat repo. Correct me if I am wrong - this is to be corrected on there and not on Zeeschuimer

**Comment by namansnghl** [comment]
Here is the URL to the issue I created
https://github.com/digitalmethodsinitiative/4cat/issues/471

**Comment by dale-wahl** [comment]
Fixed in recent updates to 4CAT: https://github.com/digitalmethodsinitiative/4cat/issues/471
